js/main.js

```diff
@@ -1,49 +1,97 @@
 /* global data */
+// variable definitions
+var $image = document.querySelector('.image');
+var $photoURL = document.querySelector('.photo-url');
+var $journalEntry = document.querySelector('form');
+var $title = document.querySelector('.title');
+var $notes = document.querySelector('.notes');
 
-var $photoURLInput = document.querySelector('#photoURL');
-var $updateImg = document.querySelector('img');
-$photoURLInput.addEventListener('input', updateURL);
-
-function updateURL(event) {
-  var target = event.target.value;
-  $updateImg.setAttribute('src', target);
+// viewswapping
+var $form = document.querySelector('.entries');
+var $entriesLink = document.querySelector('.entries-link');
+var $entries = document.querySelector('.new-entries');
+var $new = document.querySelector('.newb');
+var $save = document.querySelector('.save');
+// edit
+var $h2Edit = document.querySelector('.edit');
+var $h2New = document.querySelector('.new-entry');
+// delete
+// var $delete = document.querySelector('.delete');
+// var $modal = document.querySelector('.modal');
+// var $cancel = document.querySelector('.cancel');
+// var $confirm = document.querySelector('.confirm');
+
+// Event listeners
+$photoURL.addEventListener('input', updateImage);
+$journalEntry.addEventListener('submit', newEntry);
+// click target for editing
+$save.addEventListener('click', entriesView);
+$entriesLink.addEventListener('click', entriesView);
+$new.addEventListener('click', formView);
+// click target for delete
+// $delete.addEventListener('click', formView);
+// $cancel.addEventListener('click', handleDelete);
+// $confirm.addEventListener('click', handleDelete);
+
+// function definitons
+//
+function updateImage(event) {
+  $image.src = event.target.value;
 }
 
-var $form = document.querySelector('form');
-$form.addEventListener('submit', submitForm);
-
-function submitForm(event) {
+function newEntry(event) {
   event.preventDefault();
-  var entry = {};
-  entry.title = $form.elements.title.value;
-  entry.photo = $form.elements.photo.value;
-  entry.notes = $form.elements.notes.value;
-  entry.entryId = data.nextEntryId++;
-  data.entries.unshift(entry);
-  $ul.prepend(newEntry(data.entries[0]));
-  $form.reset();
-  $updateImg.setAttribute('src', 'images/placeholder-image-square.jpg');
-  data.view = 'entries';
-  entriesView();
+  // var $title = document.querySelector('.title');
+  // var $notes = document.querySelector('.notes');
+  var newObject = {
+    title: $title.value,
+    image: $photoURL.value,
+    notes: $notes.value,
+    id: data.nextEntryId
```

Comment:
you don't want to assign this value if its an edit because an edited entry needs to keep the same id. 

I would only assign the title, image, and notes until its determined to be a new entry or an edit, then once you know, assign the proper id, either a new one for a new entry, or the old one for an old entry